packages/babel-plugin/src/index.tsx

```diff
@@ -13,39 +14,52 @@ export default declare<State>((api) => {
     visitor: {
       Program: {
         exit(path, state) {
-          if (state.compiledImportFound && !path.scope.getBinding('React')) {
+          if (state.compiledImports && !path.scope.getBinding('React')) {
             // React is missing - add it in at the last moment!
             path.unshiftContainer('body', template.ast(`import * as React from 'react'`));
           }
         },
       },
       ImportDeclaration(path, state) {
         if (path.node.source.value === '@compiled/core') {
-          state.compiledImportFound = true;
+          // The presence of the module enables CSS prop
+          state.compiledImports = {};
+
           path.node.specifiers = path.node.specifiers
-            .filter(
-              (specifier) =>
-                specifier.local.name !== 'styled' && specifier.local.name !== 'ClassNames'
-            )
+            .filter((specifier) => {
+              if (!state.compiledImports || !t.isImportSpecifier(specifier)) {
+                return false;
+              }
+
+              if (specifier.imported.name === 'styled') {
+                state.compiledImports.styled = specifier.local.name;
+                // Remove the import
+                return false;
```

Comment:
nit: Are we concerned about loop performance? Coz we can do `.find` and `.filter` separately for readability may be. Otherwise this looks cool 🎉

Reply:
thanks!

honestly we might be concerned - we'll find out later when we roll out to Jira 😓